examples/llm/tape/tape/dataset/llm/offline/base.py

```diff
@@ -0,0 +1,119 @@
+import json
+import os
+from abc import abstractmethod
+from functools import partial
+from pathlib import Path
+from typing import Dict, List, Union
+
+from jinja2 import Environment, FileSystemLoader
+from tape.dataset.llm.engine import (
+    LlmEngine,
+    LlmOfflineEngineArgs,
+    LlmResponseModel,
+)
+from tape.dataset.parser import Article
+from tqdm import tqdm
+from transformers import AutoTokenizer
+from vllm import LLM, SamplingParams
+
+
+class LlmOfflineEngine(LlmEngine):
```

Comment:
I think there's some overlap with Rishi's current work on adding G-Retriever: https://github.com/pyg-team/pytorch_geometric/pull/9462

It would be nice to keep the LLM models in a specific directory,  as it was done by Rishi in the above PR.

+@puririshi98 for further insights

Reply:
yes i definitely agree here. as it stands this pr is basicaly the equivalent of making your own research repository on github and using PyG. there is no integration into the actual framework.  there are 32 changed files which is far too much to effectively review at once. For adding G-retriever I have broken my PR down into smaller PRs first. however before embarking on that I recommend you understand how your work could actualy be integrated into torch_geometric itself, rather than just what is essentially a standalone example repository.

Reply:
Thank you, @Kh4L and @puririshi98, for your feedback.

@puririshi98, I agree with you. There is no integration into the actual framework; hence, it is under the `examples` directory. The proposed TAPE method converts title/abstract text into LLM responses (prediction and explanation) into node features and uses it to train respective GNN models and combine the result of each GNN model. I have created a repository extending this method: [tag-llm](https://github.com/devanshamin/tag-llm).

@rusty1s, how would you like to move forward with this since you proposed it?